source/internal.d.ts

```diff
@@ -51,3 +51,17 @@ The reason a simple `keyof Union` does not work is because `keyof` always return
 @link https://stackoverflow.com/a/49402091
 */
 export type KeysOfUnion<T> = T extends T ? keyof T : never;
+
+/**
+Extracts the type of the first element of an array.
+*/
+export type ArrayHead<TArray> = TArray extends readonly [infer THead, ...unknown[]]
```

Comment:
I think these have confusing names as they sound like the same for head/tail of the array, but actually do different things. The names should probably be something like `FirstArrayElement` and `ExceptFirstArrayElement`.

Reply:
This is in line with what I said in this comment https://github.com/sindresorhus/type-fest/pull/408#discussion_r909233665

Reply:
`FirstArrayElement` seems to be the best choice as [LastArrayElement](https://github.com/sindresorhus/type-fest/blob/main/source/last-array-element.d.ts#L21) works in the same way.